examples/common.h

```diff
@@ -15,7 +15,7 @@
 
 struct gpt_params {
     int32_t seed          = -1;   // RNG seed
-    int32_t n_threads     = std::min(4, (int32_t) std::thread::hardware_concurrency());
+    int32_t n_threads     = 0;
```

Comment:
Why not just assign `n_threads` using your function? The added bonus to doing it this way is people will see the default number of cores when they call `main -h` to see the list of options.
```suggestion
    int32_t n_threads     = get_num_physical_cores();
```